### PR TITLE
chore(ci): remove gitlab-ci configuration after switch to gh actions

### PR DESCRIPTION
**Pull request by vdovhanych**
- chore(ci): remove gitlab-ci configuration after switch to gh actions
- feat(ci): separate tenv and regtest builds
- chore(ci): rename action and setup to run on trezor repository only

This pr removes Gitlab ci and updates GH action. Separates build of tenv and regtest to two actions. Sets up that actions will only run on trezor repository (mainly if someone forks this they will not be getting failed actions every night) and some changes to action names.

### CODE REVIEW

File: .github/workflows/build_and_push.yml [hunk 1]
```diff
@@ -1,5 +1,4 @@
-name: Build and push images to registry
-
+name: Bild and push trezor-user-env image
```
Comment (grdddj) [inline .github/workflows/build_and_push.yml:1]: This `bild` is a new slang-word? :D Used here at least two times

File: .github/workflows/build_and_push.yml [hunk 2]
```diff
@@ -10,6 +9,7 @@ on:
 jobs:
   build_images:
     runs-on: ubuntu-latest
+    if: ${{ github.repository_owner	== 'trezor' }}
```
Comment (grdddj) [inline .github/workflows/build_and_push.yml:12]: There seems to be a lot of whitespace at the left of `==`. Same in the file below

**Review by grdddj — COMMENTED** [review]
utACK, just some stylistic nits

**Comment by tsusanka** [comment]
Leaving this one up to you guys 👌 :).

File: .github/workflows/build_and_push.yml (hunk 1 above)
Comment (vdovhanych) [reply .github/workflows/build_and_push.yml:1]: Thanks, I missed this somehow :laughing:

**Review by grdddj — COMMENTED** [review]
Thanks for the changes. I still have some questions/clarifications about the migration:

- will `Gitlab` stop completely as soon as we merge this to `master`? There is still that automatic pipeline from `master` every 12 hours
- are we sure that no one is still using the `Gitlab` image?
- is the `GHA` setup to also build new images every day? I see `0 1 * * *` cron, which would be fine
- in this `PR` I cannot see the `Build and push test image` job - is the new `  pull_request:
    types: [opened, reopened]` working as expected?
- ii past `PR`s I think all the stages were starting at the same time - can we make it a gradual process, like in `Gitlab` - so that we will not `build and push` the test-image if the `validation checks` and `controller tests` failed?

**Comment by vdovhanych** [comment]
> Thanks for the changes. I still have some questions/clarifications about the migration:
> 
> * will `Gitlab` stop completely as soon as we merge this to `master`? There is still that automatic pipeline from `master` every 12 hours
 - Yes after we merge this gitlab won't run anything anymore.
> * are we sure that no one is still using the `Gitlab` image?
 - all projects that are using trezor-user-env should be switched to github image already
> * is the `GHA` setup to also build new images every day? I see `0 1 * * *` cron, which would be fine
 - yes it builds every night
> * in this `PR` I cannot see the `Build and push test image` job - is the new `  pull_request: types: [opened, reopened]` working as expected?
 - you're right i will only run on opened and reopend and not every push, i will fix this.
> * ii past `PR`s I think all the stages were starting at the same time - can we make it a gradual process, like in `Gitlab` - so that we will not `build and push` the test-image if the `validation checks` and `controller tests` failed?
 - good idea i will implement this here

**Comment by vdovhanych** [comment]
I set up a pipline-like workflow. @grdddj what do you think? In my opinion, it's indeed better this way.

File: .github/workflows/pull_request_ci.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+          nix_path: nixpkgs=channel:nixos-unstable
+
+      - name: Download emulator and bridge binaries
+        run: nix-shell --run "./src/binaries/firmware/bin/download.sh && ./src/binaries/trezord-go/bin/download.sh"
+      - name: "Install poetry"
+        run: nix-shell --run "poetry install"
+      - name: Run the tests
+        run: nix-shell --run "poetry run make test"
+  # Build and push test trezor-user-env image if all validation jobs pass
+  build_images:
```
Comment (grdddj) [inline .github/workflows/pull_request_ci.yml:42]: We can rename it to `build_test_images`, so it explicitly says it is `test` and not `latest`

**Comment by grdddj** [comment]
> I set up a pipeline-like workflow. @grdddj what do you think? In my opinion, it's indeed better this way.

Nice, it really ran all three steps in a pipeline, in order which we specified. ACK that.

It may be unrelated, but the `build_images ` failed because of `unable to download 'https://github.com/NixOS/nixpkgs/archive/7fad01d9d5a3f82081c00fb57918d64145dc904c.tar.gz': HTTP error 500` - https://github.com/trezor/trezor-user-env/runs/6271216570?check_suite_focus=true#logs

File: .github/workflows/pull_request_ci.yml (hunk 3 above)
Comment (grdddj) [reply .github/workflows/pull_request_ci.yml:42]: And it is actually not `images`, but just one `image` - so I suggest `build_test_image` or `build_tenv_test_image`. Also it should mention `push`, so what about `build_and_push_tenv_test_image`?

**Comment by vdovhanych** [comment]
yeah, that looks like some internal GitHub issue 😨

**Comment by grdddj** [comment]
> yeah, that looks like some internal GitHub issue fearful

Right now even the first job did not succeed - a nice test that indeed the whole pipeline is cancelled as soon as something fails

**Review by grdddj — APPROVED** [review]
LGTM!

**Pull request merged**